benchexec/tools/ultimate.py

```diff
@@ -474,11 +474,16 @@ def get_java_installations(self):
             "/usr/lib/jvm/java-*-openjdk-amd64/bin/java",
         ]
 
-        candidates = [c for entry in candidates for c in glob.glob(entry)]
+        candidates_extended = []
+        for c in candidates:
+            if "*" in c:
+                candidates_extended += glob.glob(c)
+            else:
+                candidates_extended += [c]
         pattern = r'"(\d+\.\d+).*"'
 
         rtr = {}
-        for c in candidates:
+        for c in candidates_extended:
             candidate = shutil.which(c)
```

Comment:
With your explanation, I could understand what the code is doing, but not without it. Maybe it is better to refactor this completely?

Something like `candidates = [shutil.which("java"), "/usr/bin/java"] + glob.glob("/opt/oracle-jdk-bin-*/bin/java") + ...` should make much more clear what is being done.

Reply:
I simply copied this code from Ultimate's own wrapper script `Ultimate.py` to really be sure both methods resolve the Java version in exactly the same way. We've had some issues when benchmarking recently, because depending on how you invoke benchexec, the tool info module and our wrapper script find different Java versions and java crashes with rather cryptic errors.

But if you think it's important, I could have a look at refactoring this code (in both scripts).

Reply:
Ok, I see. I can understand that you prefer it to be similar. Maybe you even want to cross-reference the code then?

While I do recommend considering the refactoring, of course this does not need to be done now.

Reply:
:+1: I've added a comment to cross-reference the wrapper script.